invenio_cli/utils.py

```diff
@@ -53,11 +62,84 @@ def destroy_containers(dev):
         subprocess.call(command)
 
 
-def cookiecutter_repo(flavor):
-    """Utility function. Returns the Cookiecutter repository of a flavour."""
-    if flavor.upper() == 'RDM':
-        return {
-                'template': 'https://github.com/inveniosoftware/' +
-                            'cookiecutter-invenio-rdm.git',
-                'checkout': 'v1.0.0a2'
-            }
+class Cookiecutter(object):
+    """Cookiecutter helper object for InvenioCLI."""
+
+    def __init__(self):
+        """Cookiecutter helper constructor."""
+        self.tmp_file = None
+        self.template = None
+
+    def repository(self, flavor):
+        """Get the cookiecutter repository of a flavour."""
+        if flavor.upper() == 'RDM':
+            repo = {
+                    'template': 'https://github.com/inveniosoftware/' +
+                                'cookiecutter-invenio-rdm.git',
+                    'checkout': 'v1.0.0a2'
+                }
+            self.template = 'cookiecutter-invenio-rdm.json'
+            return repo
+
+    def create_and_dump_config(self):
+        """Create a tmp file to store cookicutters used configuration."""
+        if not self.tmp_file:
+            self.tmp_file = \
+                tempfile.NamedTemporaryFile(mode='w+', delete=False)
+
+        config = DEFAULT_CONFIG.copy()
+        # Bug when dumping default {} it's read a string
+        config['default_context'] = None
+        config['replay_dir'] = tempfile.gettempdir()
+
+        yaml.dump(config, self.tmp_file)
+
+        return self.tmp_file.name
+
+    def get_replay(self):
+        """Retrieve cookiecutters user input values."""
+        replay_path = Path(tempfile.gettempdir()) / self.template
+
+        with open(replay_path) as replay_file:
+            return json.load(replay_file)
+
+        return {}
+
+    def remove_config(self):
+        """Remove the tmp file."""
+        if self.tmp_file:
+            self.tmp_file.close()
+
+
+# DEfining buffer size to avoid using too much memory
+BUF_SIZE = 65536  # 64 KB
+
+
+def hash_file(path_to_file):
+    """Hash file to check for consistency."""
+    sha256 = hashlib.sha256()
+
+    with open(path_to_file, 'rb') as f:
+        while True:
+            data = f.read(BUF_SIZE)
+            if not data:
+                break
+            sha256.update(data)
+
+    return sha256.hexdigest()
```

Comment:
We will be using this in a later issue I assume? Checking the hash of the files I mean.

Reply:
Not sure. It was a point to be discussed maybe next week when we tackle the release/upgrade. But I thought that maybe to check if the user changed something manually it could be checked before upgrading so it does not remove things without warning.

Reply:
This is fine for now. I agree we can talk about that next week.